src/geode/basic/database.cpp

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright (c) 2019 - 2022 Geode-solutions
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+#include <geode/basic/database.h>
+
+#include <async++.h>
+
+#include <ghc/filesystem.hpp>
+
+#include <absl/container/flat_hash_map.h>
+#include <absl/synchronization/mutex.h>
+#include <absl/time/time.h>
+
+#include <geode/basic/identifier.h>
+#include <geode/basic/pimpl_impl.h>
+#include <geode/basic/uuid.h>
+
+namespace
+{
+    constexpr auto DATA_EXPIRATION = absl::Seconds( 10 );
```

Comment:
To discuss, 10s are for testing. A production value could be 5min?

Reply:
When the data is not used for this amount of time, it is unloaded from the memory.